### PR TITLE
Update msg_stack.py

### PR DESCRIPTION
**Pull request by yangpryili**
1、[msg.to_dict() for msg in msgs], msg is not instanse of Message, it not has a func of to_dict(), so msg.to_dict() will accur JSONDecodeError; 
2、json.dump(new_data, file), it appends new_data to the end of the file instead of overwriting from the beginning, Hence, it's necessary to first perform file.seek(0) and file.truncate().

### CODE REVIEW

File: opendevin/server/session/msg_stack.py [hunk 1]
```diff
@@ -99,7 +99,11 @@ async def _del_messages(self, del_sid: str):
                 new_data = {}
                 for sid, msgs in data.items():
                     if sid != del_sid:
-                        new_data[sid] = [msg.to_dict() for msg in msgs]
+                        new_data[sid] = msgs
+                # 移动文件指针到文件开头以便覆盖原内容
+                file.seek(0)
+                # 清空文件原有内容
+                file.truncate()
```
Comment (yufansong) [inline opendevin/server/session/msg_stack.py:106]:
```suggestion
                        new_data[sid] = msgs
                # Move the file pointer to the beginning of the file to overwrite the original contents
                file.seek(0)
                # clean previous content
                file.truncate()
```

**Review by yufansong — APPROVED** [review]
Change comments into English. Others LGTM.

**Comment by SmartManoj** [comment]
Solves https://github.com/OpenDevin/OpenDevin/issues/1809

**Comment by SmartManoj** [comment]
> so msg.to_dict() will accur JSONDecodeError;

will raise AttributeError: 'dict' object has no attribute 'to_dict'

**Pull request merged**